js/src/utils/concatenateListOfWords.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * A function that joins a list of words using the sepators provided as the second and third argument.
+ *
+ * Example:
+ *
+ * concatenateListOfWords(['A', 'B', 'C']) will return 'A, B and C'
+ *
+ * @param  {Array<string>} listWords A list of words to concatenate.
+ * @param  {string} [separator] The separator between words.
+ * @param  {string} [separatorLastElement] The separator for the last word.
+ * @return {string} The concatenated string.
+ */
+
+export default function concatenateListOfWords(
+	listWords,
+	separator = ', ',
+	separatorLastElement = 'and'
+) {
+	if ( listWords.length === 1 ) {
+		return listWords.slice( -1 ).join( '' );
+	}
+	return `${ listWords
+		.slice( 0, -1 )
+		.join( separator ) } ${ separatorLastElement } ${ listWords.slice(
+		-1
+	) }`;
```

Comment:
💅 or ❔ 
> There is some complexity in adding the singular and plural for error messages, for example:
> 
> `There **are** errors in the following action**s**` `There **is an error** for the following **action**`
> 
> I think this could be improve in a different PR and I am happy to hear about tools or functions that could help us to improve this part.

There is the `_n` in `@wordpress/i18n` package but it doesn't perfectly fit with this implementation. If it's ok to internalize this handling to `useCreateNoticeForMultipleErrors`, it could be replaced with

```js
const listErrors = sprintf(
	_n(
		'There is an error in the following action: %1$s%2$s',
		'There are errors in the following actions: %1$s and %2$s',
		listWords.length,
		'google-listings-and-ads'
	),
	listWords.slice( 0, -1 ).join( separator ),
	listWords.at( -1 )
);
```